test/cpp/common/channel_arguments_test.cc

```diff
@@ -35,10 +35,50 @@
 
 #include <grpc/grpc.h>
 #include <gtest/gtest.h>
+#include "src/core/lib/iomgr/socket_mutator.h"
 
 namespace grpc {
 namespace testing {
 
+namespace {
+
+// A simple grpc_socket_mutator to be used to test SetSocketMutator
+class TestSocketMutator : public grpc_socket_mutator {
+ public:
+  TestSocketMutator();
+
+  bool MutateFd(int fd) {
+    // Do nothing on the fd
+    return true;
+  }
+};
+
+//
+// C API for TestSocketMutator
+//
+
+bool test_mutator_mutate_fd(int fd, grpc_socket_mutator* mutator) {
+  TestSocketMutator* tsm = (TestSocketMutator*)mutator;
+  return tsm->MutateFd(fd);
+}
+
+void test_mutator_destroy(grpc_socket_mutator* mutator) {
+  TestSocketMutator* tsm = (TestSocketMutator*)mutator;
+  delete tsm;
+}
+
+grpc_socket_mutator_vtable test_mutator_vtable = {test_mutator_mutate_fd,
```

Comment:
Thanks a lot for the review!
The cmp function in `grpc_arg_pointer_vtable` is implemented in [socket_mutator.c](https://github.com/grpc/grpc/pull/8335/files#diff-66f165bfbf516848b03e99c9bedd5254R69). It compares the pointers using `GPR_ICMP()`.
`grpc_socket_mutator_vtable` does not have a cmp function. Should we add one?